.clang-tidy

```diff
@@ -23,7 +23,7 @@ Checks: '*,
     -bugprone-implicit-widening-of-multiplication-result,
     -bugprone-narrowing-conversions,
     -bugprone-not-null-terminated-result,
-    -bugprone-reserved-identifier,
+    -bugprone-reserved-identifier, # useful but too slow, TODO retry when https://reviews.llvm.org/rG1c282052624f9d0bd273bde0b47b30c96699c6c7 is merged
```

Comment:
@rschu1ze `Checks` is actually a string, so you can't use comments here:

```
$ clang-tidy -verify-config
/src/ch/clickhouse/.clang-tidy: warning: unknown check '# useful but too slow' [-verify-config]
/src/ch/clickhouse/.clang-tidy: warning: unknown check 'TODO retry when https://reviews.llvm.org/rG1c282052624f9d0bd273bde0b47b30c96699c6c7 is merged
    -bugprone-unchecked-optional-access' [-verify-config]
/src/ch/clickhouse/.clang-tidy: warning: unknown check 'cppcoreguidelines-avoid-const-or-ref-data-members' [-verify-config]
/src/ch/clickhouse/.clang-tidy: warning: unknown check 'cppcoreguidelines-avoid-do-while' [-verify-config]
/src/ch/clickhouse/.clang-tidy: warning: unknown check '# useful but slooow
    -misc-use-anonymous-namespace' [-verify-config]
/src/ch/clickhouse/.clang-tidy: warning: unknown check '# useful but too slow
    -readability-implicit-bool-conversion' [-verify-config]
/src/ch/clickhouse/.clang-tidy: warning: check glob 'zirkon-*' doesn't match any known check [-verify-config]
/src/ch/clickhouse/.clang-tidy: warning: unknown check option 'readability-identifier-naming.TemplateUsingCase' [-verify-config]
/src/ch/clickhouse/.clang-tidy: warning: unknown check option 'readability-identifier-naming.UsingCase'; did you mean 'readability-identifier-naming.UnionCase' [-verify-config]
/src/ch/clickhouse/.clang-tidy: warning: unknown check option 'cppcoreguidelines-avoid-do-while.IgnoreMacros' [-verify-config]
```

Reply:
Interesting, didn't know about `-verify-config` ...

Tbh, being able to write comment behind checks is just so damn useful that I'd prefer to see some warnings (they don't break the build whatsoever). I can fix the other warnings though.